druid/src/widget/image.rs

```diff
@@ -131,14 +131,10 @@ impl ImageData {
 
     /// Load an image from a DynamicImage from the image crate
     pub fn from_dynamic_image(image_data: image::DynamicImage) -> ImageData {
-        match image_data.color() {
-            image::ColorType::RGBA(_) | image::ColorType::BGRA(_) | image::ColorType::GrayA(_) => {
-                Self::from_dynamic_image_with_alpha(image_data)
-            }
-            image::ColorType::RGB(_)
-            | image::ColorType::Gray(_)
-            | image::ColorType::Palette(_)
-            | image::ColorType::BGR(_) => Self::from_dynamic_image_without_alpha(image_data),
+        match image_data.color().channel_count() {
```

Comment:
I sort of really like this.
I also really dont. 

If we bump the version of image in the future and they introduce a new type that dose not obay this rule it might be hard to debug with the new change but if we try to update image with the old code then rust will warn us that a new type has been added and we can think about what were are doing.
The idea would be for /channel_count/alpha_present/ I will take a action to go make a issue with the image crate for this.

But the commit message dose not at all match this change, I know druid is much less worried about our history than I normally am so I'm not sure that should block this.

But if it was in a separate PR then we could discus this while not blocking most of this PR

Reply:
Actioned https://github.com/image-rs/image/issues/1167

Reply:
yea, this change was a result of updating our `image` dependency; it broke its API to rename a bunch of things in `ColorType`, and when I went and looked at the API this was the solution that seemed simplest. I'll acknowledge it's a bit gross. 😒

Reply:
@fishrockz perhaps you can open a new issue for us here too to track this? Then we can get the version updates done and deal with the robustness of the `Image` widget separately.

Reply:
Do we need to bump image now? I am happy to create a issue bump image once its fixed? but if we have a stonge reason to want a newer version of image for 0.0.5 then i dont think this should block.

Reply:
I'll do a more robust impl here, no worries

Reply:
follow up issue when the image crate people get back to us, https://github.com/xi-editor/druid/issues/748

Reply:
okay, I updated this. You can no longer ensure you're matching all items, because the enum is non-exhaustive, but it should be more robust.

Reply:
Apart from @xStrom's comment. I am very happy with this. in gitlab i would click resolve but i think think github has a concept of unresolved threads.

Reply:
Threads can be resolved, but merging isn't blocked here due to unresolved threads. Resolving hides the comments though, so I tend to only do it for trivial things.